### DISCUSSION TITLE
Upgrade from 7 to @latest: If the file is ignored because it is located outside of the base path, change the location of your config file to be in a parent directory

### DISCUSSION THREAD

**Discussion by disja** [eslint/eslint#18806] [General]
Hi, I've recently updated from version 7 to the latest version of ESLint. Unfortunately, I've encountered an issue that I haven't been able to resolve yet, and I'd appreciate any help or ideas. I should also mention that my project has some unique requirements.

Here’s my project structure:

![image](https://github.com/user-attachments/assets/ce10b2c8-8911-47ec-9873-4baf626031bf)

I need to lint only the files in the main folder, and this setup cannot be changed. In version 7, I could simply run eslint path in the CLI, and everything worked fine. However, since updating to version 9, this approach no longer works. Instead, I receive the following message: "If the file is ignored because it is outside the base path, change the location." Unfortunately, changing the location is not an option for me.

One more thing to note: this project is not a Single Page Application (SPA) but a multipage setup.

Any assistance would be greatly appreciated :)

**Comment by nzakas** [comment]
At the moment, your only option is to change the directory to `test` before running ESLint:

```sh
cd test && eslint .
```

We are working on a change that will restore the previous behavior:
https://github.com/eslint/eslint/pull/18742

**Reply by fljotavi** [reply]
Hi Nicholas, I think your PR addresses a different problem.

OP's problem is that files outside of base path are being ignored. If you run `cd test && eslint`, files under the `/main` folder are still being ignored because the base path of the configuration file is `/test`.

#18742 doesn't essentially alter this logic. The code that's relevant to OP's concern seems to be the match functions in eslint helpers (see below). It now still ignores all files outside of the base path, which I believe is not pre-V9 behavior.

https://github.com/eslint/eslint/blob/522d8a32f326c52886c531f43cf6a1ff15af8286/lib/eslint/eslint-helpers.js#L374
https://github.com/eslint/eslint/blob/522d8a32f326c52886c531f43cf6a1ff15af8286/lib/eslint/eslint-helpers.js#L161

I think there must be a reason it's designed to work this way. Regardless, do you have a suggested practice for cases like this where you need to lint an out-of-package file?

**Reply by nzakas** [reply]
> OP's problem is that files outside of base path are being ignored. If you run `cd test && eslint`, files under the `/main` folder are still being ignored because the base path of the configuration file is `/test`.

That's how every CLI tool works, though:

```shell
$ cd test

# this means "run eslint in the test directory"
$ npx eslint
```

If you want to lint files in `main`, then  you need to run ESLint from the root of the project and then run:

```shell
$ npx eslint main
```

Then you'd just need to add a config file in `main` or in the root of the project.

**Reply by fljotavi** [reply]
It's not always possible to `cd` to the parent directory. In my case, the `eslint` command is run by a build tool that I don't have direct control over. OP probably faced similar scenarios.

Also, I think it's not uncommon for CLI tools to support running on files outside of the current directory. That's how ESLint 8 worked, and it's quite handy when I need to also lint some shared JS utils from adjacent directories along with the current project.

**Comment by SaiCode-OTH** [comment]
Okay. 9.12.0 has merged it but 
```bin/eslint --flag unstable_config_lookup_from_file main/ --fix```
causes:
> ESLint couldn't find an eslint.config.(js|mjs|cjs) file.

`bin/eslint --flag unstable_config_lookup_from_file -c "test/eslint.config.mjs" main/ --fix`
causes: 
> warning  File ignored because outside of base path

Which is still not pre v9 behaviour.

**Reply by mdjermanovic** [reply]
@SaiCode-OTH can you please open a bug report with all the details?

**Reply by laoneo** [reply]
Is there an issue for this? Because in v8 you could lint files outside of folder where the config is in, while in v9 it doesn't work. Yes, I tried with the `unstable_config_lookup_from_file` flag.

**Reply by HolgerJeromin** [reply]
#19118 is probably a matching issue.

**Reply by laoneo** [reply]
I made the workaround to change the directory and then do the linting from there. But would love to see the old behavior.

**Comment by Peredery** [comment]
@mdjermanovic facing similar issue:

I have config file inside: `/app/eslint.config.mjs` and want to lint files inside `/tmp/example.js`

```
root@container:/app  npx eslint --flag unstable_config_lookup_from_file -c "./eslint.config.mjs" /tmp/example.js --fix

/tmp/example.js
  0:0  warning  File ignored because outside of base path
 ```

**Reply by nzakas** [reply]
Can you open an issue with a reproduction case so we can take a look?

**Reply by HolgerJeromin** [reply]
his issue #19118 has a repro case

**Comment by andrewevans** [comment]
Since my server files are in a child directory of my main project, AND where all my npm node modules and their `node_modules/` directory is, I call it _from the project root_ manually like `./MY_SERVER_DIRECTORY/node_modules/eslint/bin/eslint.js -c MY_SERVER_DIRECTORY/eslint.config.mjs MY_FILE.js` . This 1) runs eslint, 2) with its custom config, 3) at the file specified relative to where I'm calling it. My project looks like this:

- project root
- - MY_SERVER_DIRECTORY
- - - `node_modules` with eslint
- - - eslint.config.mjs
- MY_FILE.js

The trick to remember is that all paths are relative _to where you call it_. My odd setup is so that my project root is only the project files and none of the optional server/debug/dev stuff. But nonetheless this illustrates how to call eslint from a directory outside of where it is normally run when you run it from the command like `> eslint MY_FILE.js` or whatever your command is.
